### PR TITLE
remove code not strictly required for ZQL, update comments [1]

### PR DESCRIPTION
**Pull request by tantaman**
- Remove `Queue`. We never actually build up a queue of values in this implementation. Everything is batched into a single multiset each transaction due to our `queue`, `notify`, `run` staging.
- Remove `PersistentTreeView`  (we only need a mutable tree for ZQL. For now anyway..)
- Remove `AbstractTreeView` (no longer have two different tree views since we removed the persistent one)
- Remove `IDifferenceStream` (ZQL only has a single implementation)
- Remove `PassThroughOperator` (ZQL doesn't need to temporarily detach pipelines)
- Remove `StatelessSource` (ZQL always queries history)

[GroupBy](https://github.com/rocicorp/rails/tree/mlaw/group-by) has some changes to aggregation that will let us:
- Remove `AbstractView`
- Remove `PrimitiveView`

The `queue`, `notify`, `run`, `notifyCommitted` pattern still exists. It doesn't seem _that_ difficult to grok once I looked at it again to consider removing it.

### CODE REVIEW

File: src/zql/ast-to-ivm/pipeline-builder.test.ts [hunk 1]
```diff
@@ -15,10 +15,11 @@ const e1 = z.object({
 type E1 = z.infer<typeof e1>;
 
 const context = makeTestContext();
+const comparator = (l: E1, r: E1) => l.id.localeCompare(r.id);
```
Comment (arv) [inline src/zql/ast-to-ivm/pipeline-builder.test.ts:18]: FYI. localeCompare is not the right thing to use but for tests 🤷🏼

File: src/zql/context/context.ts [hunk 2]
```diff
@@ -13,7 +20,10 @@ export function makeTestContext(): Context {
   const sources = new Map<string, Source<unknown>>();
   const getSource = <T extends Entity>(name: string) => {
     if (!sources.has(name)) {
-      sources.set(name, materialite.newStatelessSource<T>());
+      sources.set(
+        name,
+        materialite.newSetSource<T>((l, r) => l.id.localeCompare(r.id)),
```
Comment (arv) [inline src/zql/context/context.ts:25]: Use compare-utf8

https://www.npmjs.com/package/compare-utf8

Comment (arv) [reply src/zql/context/context.ts:25]: or at least non locale compare

File: src/zql/ivm/graph/operators/operator.ts [hunk 3]
```diff
@@ -45,7 +45,16 @@ export interface Operator {
 }
 
 export class NoOp implements Operator {
-  run(_version: Version) {}
+  readonly #inputs;
```
Comment (arv) [inline src/zql/ivm/graph/operators/operator.ts:48]: I know TS can infer the type for this but can we include it for readability?

File: src/zql/context/context.ts (hunk 2 above)
Comment (tantaman) [reply src/zql/context/context.ts]: added compare-utf8

**Pull request merged**